### PR TITLE
source code for bealdung article "Find IP address of client"

### PR DESCRIPTION
**Pull request by mainak9042**
source code for a server and a client application for the baeldung article

### CODE REVIEW

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/timeout/TcpServerSocket.java [hunk 1]
```diff
@@ -3,6 +3,10 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
+<<<<<<< HEAD
+//A Java program for a Server
+=======
+>>>>>>> upstream/master
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-networking-3/src/main/java/com/baeldung/timeout/TcpServerSocket.java:9]: Note: there's a strange bit of git merge conflict here that needs resolving.

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/clientaddress/ApplicationServer.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                    out.println("Hello Client !!");
+                } else {
+                    out.println("Not valid message !! ");
+                    break;
+                }
+            }
+            closeIO();
+            stopServer();
+        } catch (IOException e) {
+
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-networking-3/src/main/java/com/baeldung/clientaddress/ApplicationServer.java:44]: empty catch block. How about we remove the try/catch and make the method `throws IOException`?

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]
The reader has no way of running this code if they download it.

Can we add an IntegrationTest that runs it? or a `main` method for them to use?

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/timeout/TcpServerSocket.java (hunk 1 above)
Comment (mainak9042) [reply core-java-modules/core-java-networking-3/src/main/java/com/baeldung/timeout/TcpServerSocket.java]: i don't know what happened here. This is from my previous article. I removed the comment

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/clientaddress/ApplicationServer.java (hunk 2 above)
Comment (mainak9042) [reply core-java-modules/core-java-networking-3/src/main/java/com/baeldung/clientaddress/ApplicationServer.java]: done

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/clientaddress/ApplicationServer.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+public class ApplicationServer {
+
+    private ServerSocket serverSocket;
+    private Socket conncetedSocket;
+    private PrintWriter out;
+    private BufferedReader in;
+
+    public void startServer(int port) throws IOException{
+        try {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-networking-3/src/main/java/com/baeldung/clientaddress/ApplicationServer.java:19]: No catch block

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/clientaddress/ApplicationServer.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            serverSocket = new ServerSocket(port);
+            conncetedSocket = serverSocket.accept();
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-networking-3/src/main/java/com/baeldung/clientaddress/ApplicationServer.java:21]: Typo - should be `connectedSocket`

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

**Pull request merged**